src/opera/test/pge/test_metfile.py

```diff
@@ -0,0 +1,214 @@
+        Initialize regular expression
+        Initialize other class variables
+
+        """
+        cls.starting_dir = abspath(os.curdir)
+        cls.test_dir = resource_filename(__name__, "")
+        cls.data_dir = join(cls.test_dir, "data")
+
+        os.chdir(cls.test_dir)
+
+        cls.working_dir = tempfile.TemporaryDirectory(
+            prefix="test_time_", suffix='temp', dir=os.curdir)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        """
+        At completion re-establish starting directory
+        -------
+        """
+        cls.working_dir.cleanup()
+        os.chdir(cls.starting_dir)
+
+    def setUp(self) -> None:
+        """
+        Use the temporary directory as the working directory
+        -------
+        """
+        os.chdir(self.working_dir.name)
+
+    def tearDown(self) -> None:
+        """
+        Return to starting directory
+        -------
+        """
+        os.chdir(self.test_dir)
+
+    def testMetFile(self):
+        """
+        Tests instantiation of MetFile class
+        Methods tested in MetFile class:
+            set_key_value()
+            write_met_file()
+            read_met_file()
+
+        """
+        met_file = MetFileTestCase.TEST_MET_FILE
+        met_data = MetFile(met_file)
+        self.assertIsInstance(met_data, MetFile)
+
+        # Test __setitem__ and __getitem__
+        met_data["test key"] = "test value"
+        self.assertEqual(met_data["test key"], "test value")
+        # Test write_met_file()
+        met_data.write_met_file()
+        # Copy contents of the file into a variable
+        with open(met_file, 'r') as mf:
+            lines = (mf.readlines())
+        # Test string to look for string in the dictionary
+        test_str = '  "test key": "test value",\n'
+        self.assertIn(test_str, lines)
+        # Test read_met_file()
+        met_dict = met_data.read_met_file()
+        self.assertEqual(met_data['test key'], 'test value')
+
+        # Read another line to test for an existing file
+        met_dict = "test key 2", "test value 2"
+        met_data.write_met_file()
+        with open(met_file, 'r') as mf:
+            lines = (mf.readlines())
+        # Test string to look for
+        test_str = '  "test key 2": "test value 2"\n'
+        self.assertIn(test_str, lines)
+        # Test read_met_file()
+        met_dict = met_data.read_met_file()
+        self.assertEqual(met_dict['test key 2'], 'test value 2')
+
+    def test_validate_json_file(self):
+        """
+        Instantiates pge objects
+        Verifies that the catalog metadata json file is created
+        Verifies that the schema is correct
+        Verifies that an Error is written to the log file is there is a schema error in the catalog metadata file.
+
+        """
+        # instantiate a pge object
+        runconfig_path = join(self.data_dir, 'test_base_pge_config.yaml')
+
+        pge = PgeExecutor(pge_name='BasePgeTest', runconfig_path=runconfig_path)
```

Comment:
Rather than start from executing a full `PgeExecutor`, this test could start from creating local `MetFile` objects, both valid and invalid, and just call their validation methods on them.

Reply:
Now localizing the test to just the validate_json_file() method.